### ISSUE TITLE
[Feature Request]: An option to use text JSONS for FTB Quests

### ISSUE THREAD

**Issue by aaronhowser1** [FTBTeam/FTB-Mods-Issues#495]
### Mod

FTB Quests

### Description

[`/tellraw`-style JSONs](https://minecraft.fandom.com/wiki/Raw_JSON_text_format) are a lot more versatile than what can be done currently. Hover text, clickable text, fonts, etc.

It can even do things like translations and colors, so no lost functionality

**Comment by desht** [comment]
I like this idea, but it's likely going to be non-trivial to add.  Will need to go through and locate any places in the mod where strings are converted to Components and determine if raw JSON is appropriate in each place.  Will likely extend into FTB Library too.

**Comment by desht** [comment]
Update: easier than I thought... next build will allow use of raw JSON for the quest subtitle and description lines.  Including clickEvent and hoverEvent.

**Comment by aaronhowser1** [comment]
You are a god

**Comment by aaronhowser1** [comment]
It would be nice if we could get array support as well, so we could directly copy from sites like https://www.minecraftjson.com/

For example, the command:
```
/tellraw @p ["",{"text":"This text is bold. ","bold":true},{"text":"This text has hover text","hoverEvent":{"action":"show_text","contents":["This is hover text"]}}]
```
![image](https://user-images.githubusercontent.com/15698722/197356810-5244a2a3-8dc9-45d1-bca6-3a848f0d6644.png)

`["",{"text":"This text is bold. ","bold":true},{"text":"This text has hover text","hoverEvent":{"action":"show_text","contents":["This is hover text"]}}]` doesn't work as the first character isn't a {

Swapping the [] to {} `{"",{"text":"This text is bold. ","bold":true},{"text":"This text has hover text","hoverEvent":{"action":"show_text","contents":["This is hover text"]}}}` results in this:
![image](https://user-images.githubusercontent.com/15698722/197356863-5d4f2c2a-add4-4603-93cd-58338f254e0b.png)

So basically, we can only have one format per paragraph. The entire paragraph can be a different font, or have hover text, or any combination thereof, but it's all uniform.

**Comment by aaronhowser1** [comment]
`{"text":"Test","bold":true} {"text":"Test","italic":true,"underlined":true}` results in just the first showing:
![image](https://user-images.githubusercontent.com/15698722/197357649-f61cdfdd-6ee2-406b-9574-1d656c5aa07b.png)

**Comment by desht** [comment]
Yeah, I'll get array support into the next build.  Forgot about that.

Your second example isn't valid rawtext.  IIRC that would be correctly written as:
```json
{"text":"Test","bold":true, "extra": {"text":"Test","italic":true,"underlined":true} }
```

**Comment by desht** [comment]
Implemented for 1.18 & 1.19
